### ISSUE TITLE
Strict type checking rules violation

### ISSUE THREAD

**Issue by funivan** [kalessil/phpinspectionsea#70]
It seems to be a bug.
```php
<?
  declare(strict_types = 1);

  class Telephone {
    public function __construct(string $number) {
    }
  }

  $data = [];
  $tel = $data['number']??null;
  if ($tel === null) {
    throw new \InvalidArgumentException('Invalid data');
  }
  new Telephone($tel);  // <-- Error: Expect to be string mixed|null given
```

p.s. I know how to rewrite this code using `isset`.

**Comment by kalessil** [comment]
Is this reported already or we are talking about a FR?

**Comment by funivan** [comment]
Just a report.

**Comment by kalessil** [comment]
Then it's from IDE itself.
